lib/bundler/definition.rb

```diff
@@ -200,8 +200,10 @@ def resolve
       @resolve ||= begin
         last_resolve = converge_locked_specs
         if Bundler.settings[:frozen] || (!@unlocking && nothing_changed?)
+          Bundler.ui.debug("Resolved from lockfile")
```

Comment:
👍 Thanks!  I'll be looking at this in the next few days.  I really want to track this thing down.